src/test/cdk/wizards/previewStateMachineCDKWizard.test.ts

```diff
@@ -0,0 +1,195 @@
+    new FakeParentNode('cdkJsonPath'),
+    'MyTopLevelNode',
+    vscode.TreeItemCollapsibleState.Collapsed,
+    mockTopLevelConstructTreeEntity
+)
+
+const mockStateMachineConstructTreeEntity: ConstructTreeEntity = {
+    id: 'MyStateMachine',
+    path: 'aws-tester/MyStateMachine',
+    children: {
+        'Resource': {
+            id: 'Resource',
+            path: 'aws-tester/MyStateMachine/Resource',
+            attributes: {
+                "aws:cdk:cloudformation:type": 'AWS::StepFunctions::StateMachine'
+            }
+        }
+    }
+}
+
+const mockStateMachineNode = new ConstructNode(
+    new FakeParentNode('cdkJsonPath'),
+    'MyStateMachine',
+    vscode.TreeItemCollapsibleState.Collapsed,
+    mockStateMachineConstructTreeEntity
+)
+
+function getTestNode(): app.CdkAppLocation {
+    const mockUri = sandbox.createStubInstance(vscode.Uri)
+    sandbox.stub(mockUri, 'fsPath').value(workspaceFolderPath)
+    const mockWorkspaceFolder: vscode.WorkspaceFolder = { uri: mockUri, index: 0, name: workspaceFolderName }
+    const appLocation: app.CdkAppLocation = {
+        cdkJsonPath: 'cdkJsonPath',
+        treePath: 'treePath',
+        workspaceFolder: mockWorkspaceFolder,
+    }
+
+    return appLocation
+}
+
+describe('PreviewStateMachineCDKWizard', async function () {
+    const testNode = getTestNode()
+
+    const CDK_APPLOCATIONS: CdkAppLocationPickItem[] = []
+    CDK_APPLOCATIONS.push({
+        label: getCDKAppWorkspaceName(testNode.cdkJsonPath),
+        cdkApplocation: testNode
+    })
+
+    const TOP_LEVEL_NODES: TopLevelNodePickItem[] = []
+    TOP_LEVEL_NODES.push({
+        label: mockTopLevelNode.label,
+        topLevelNode: mockTopLevelNode
+    })
+
+    const STATE_MACHINES: ConstructNodePickItem[] = []
+    STATE_MACHINES.push({
+        label: mockStateMachineNode.label,
+        stateMachineNode: mockStateMachineNode
+    })
+
+    beforeEach(function () {
+        sandbox = sinon.createSandbox()
+    })
+
+    it('exits when cancelled', async function () {
+        const mockUserPrompt: any = () => Promise.resolve(undefined)
+        const wizard = new PreviewStateMachineCDKWizard(mockUserPrompt)
+        const result = await wizard.run()
+
+        assert.ok(!result)
+    })
+
+    it('returns undefined when cdk application does not exist', async function () {
+        const promptResults = [
+            [{
+                label: '',
+                cdkApplocation: undefined
+            }],
+            [{
+                label: '',
+                topLevelNode: undefined
+            }],
+            [{
+                label: '',
+                stateMachineNode: undefined
+            }]
+        ]
+        const mockUserPrompt: any = (options: any) => Promise.resolve(promptResults.shift())
+        const wizard = new PreviewStateMachineCDKWizard(mockUserPrompt)
+        const result = await wizard.run()
+
+        assert.deepStrictEqual(result, undefined)
```

Comment:
General comment through this file: Just use `strictEqual` for comparisons to non-objects.